test/integration/connect/envoy/case-terminating-gateway-deleted-resolver/capture.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```

Comment:
It should be easier to test this [here](https://github.com/hashicorp/consul/blob/main/agent/proxycfg/state_test.go) rather than making an entirely new integration test.

Reply:
ah, cool, I was trying to figure out where I could do something that was a little less heavy-handed, but wasn't sure where we were doing unit-test like stuff for the proxycfg structs.

Reply:
@erichaberkorn removed the integration test and added a state-based unit test that does the job. Thanks for the tip!